CMakeLists.txt

```diff
@@ -16,23 +16,48 @@ option(ENABLE_NESO_TESTS
 include(cmake/Sanitizers.cmake)
 include(cmake/CheckFileList.cmake)
 
+
+
 # ##############################################################################
 # Find dependencies
 # ##############################################################################
-find_package(SYCL REQUIRED)
+
+#USE -DNESO_SYCL_VENDOR to chose implementation at gen time
+set(NESO_SYCL_VENDOR "Default" CACHE STRING "SYCL vendor to look for - required
+if present")
+include(cmake/SYCL.cmake)
+if (NESO_SYCL_VENDOR MATCHES "Default")
+    find_sycl()
+else()
+    find_sycl(${NESO_SYCL_VENDOR})
+endif()
 if(SYCL_FOUND)
-  message(STATUS "Using SYCL implementation ${SYCL_IMPLEMENTATION}")
-  if(hipSYCL_FOUND)
-    set(HIPSYCL_TARGETS "omp")
-    set(SYCL_FLAG NESO_HIPSYCL)
-  elseif(SYCL_IMPLEMENTATION STREQUAL "IntelDPCPP")
-    set(SYCL_FLAG NESO_DPCPP)
-  endif()
+    message(STATUS "Using SYCL implementation ${SYCL_IMPLEMENTATION}")
+    if(NOT SYCL_INTEL)
+        set(HIPSYCL_TARGETS "omp")
+        set(ACPP_TARGETS "omp")
+        set(SYCL_FLAG NESO_HIPSYCL)
+    else()
+        set(SYCL_FLAG NESO_DPCPP)
+    endif()
```

Comment:
Presumably this should match whatever you and @will-saunders-ukaea settle on in the [NESO-Particles pull request](https://github.com/ExCALIBUR-NEPTUNE/NESO-Particles/pull/57#discussion_r1515961915).